app/models/game.rb

```diff
@@ -40,4 +40,9 @@ def populate_board!
     King.create(:game_id => self.id, :color => "White", :row_position => 7, :col_position => 4)
     King.create(:game_id => self.id, :color => "Black", :row_position => 0, :col_position => 4)
   end
+
+  def occupied?(x, y)
+    Piece.where(:row_position => x, :col_position => y).present?
```

Comment:
pieces.where
Do you see the difference?  Imagine there are 2 or more games in the database.